num.nix

```diff
@@ -38,62 +43,105 @@ rec {
     then y
     else x;
 
-  mod = base: int: base - (int * (builtins.div base int));
+  /* compare :: number -> number -> int
 
-  /* even :: integer -> bool
+     Compares two numbers and returns -1 if the first is less than the second, 0
+     if they are equal, or 1 if the first is greater than the second.
   */
-  even = x: mod x 2 == 0;
+  compare = x: y:
```

Comment:
do you think we should instead return a string \elem { "EQ", "GT", "LT" }?

Reply:
probably, but I'd love to have a system for defining enums.

should I just write the type as `number -> number -> "EQ" | "GT" | "LT"` or something?

Reply:
with respect to the first comment, it would be nice if there was an easy way to `match` on the result, for example. such enums could be automatically generated, and would come for free if we had a system for defining ADTs, like in https://github.com/shlevy/nix-adt for example.

Reply:
alright I'm sticking to just writing the type the typescript way